Add release notes for domain object collection finalization (#36919)
gradle/gradle: platforms/documentation/docs/src/docs/release/notes.md

}
```

### Domain Object Collections can be made immutable

You can now make domain object collections immutable with `disallowChanges()` and you can check on the mutability of a collection with `areChangesAllowed()`.  This allows plugin and build authors to ensure that the contents of a collection are now final and cannot be further modified.  

> Note that this does not prevent modifications to the individual elements, only that elements cannot be added or removed once this call is invoked.  

Reviewer: Can we turn this into a more readable bullet point list?

```suggestion
- Once `disallowChanges()` is called, elements can no longer be added to or removed from the collection.
```
